### DISCUSSION TITLE
MeshCentral won't use MariaDB connection

### DISCUSSION THREAD

**Discussion by phpmaven** [Ylianst/MeshCentral#3926] [General]
Hi All,

So I'm trying to get MeshCentral to use my local MariaDB server and although it will connect, it won't actually ever use the database that I designate for it. I have this in my config.json:

```
"mysql": { 
      "host": "localhost",
      "user": "my_user_name", 
      "port": 3306,
      "password": "my_password",
      "database": "meshcentral"
 }
```

I know it connects because if I put the wrong password in it will throw an error on startup.
Once it starts if I add a new remote computer it will show up but I can't connect to it and it disappears as soon as I refresh the "My Devices" page.
This is on Centos

Is there any way to figure out why it's not actually adding tables to the DB and using it?

Thanks,

Mark

P.S. Thank you so much Ylian for creating and maintaining this wonderful tool.

**Comment by phpmaven** [comment]
Turns out it was because my MariaDB was woefully out of date. Everything is good now. Probably wouldn't be a bad idea to add a minimum version check to db.js right after it connects.

Mark

**Comment by Ylianst** [comment]
Oh good, thanks for reporting back. Using `"mysql"` to connect to a MariaDB should work fine, but if you have any issues, you can use the `"mariadb"` configuration section in the config.json. It's in the [schema here](https://github.com/Ylianst/MeshCentral/blob/1c2dc30d8e0c21f8e7730aaad8e649e3be767d8f/meshcentral-config-schema.json#L15).

That said, MariaDB is plenty compatible with MySQL so should not be an issue.

**Comment by phpmaven** [comment]
Hi  Ylian, everything is working great.

Thank you,
Mark Langdon
Tech Mavens
